slack-clone-compose-sample/src/main/kotlin/io/getstream/slack/compose/ui/channels/ChannelsScreen.kt

```diff
@@ -0,0 +1,194 @@
+    listViewModel: ChannelListViewModel,
+    workspace: Workspace,
+    onItemClick: (Channel) -> Unit = {}
+) {
+    val currentUser by listViewModel.user.collectAsState()
+    val isNetworkAvailable by listViewModel.isOnline.collectAsState()
+
+    var searchQuery by rememberSaveable { mutableStateOf("") }
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(ChatTheme.colors.appBackground)
+    ) {
+
+        ChannelListHeader(
+            modifier = Modifier.height(56.dp),
+            currentUser = currentUser,
+            title = workspace.title,
+            isNetworkAvailable = isNetworkAvailable,
+            leadingContent = { WorkspaceLogo(logo = workspace.logo) },
+            titleContent = {
+                WorkspaceTitle(
+                    modifier = Modifier.weight(1f),
+                    title = workspace.title
+                )
+            },
+            trailingContent = { Spacer(Modifier.width(36.dp)) },
+        )
+
+        Card(
+            modifier = Modifier
+                .padding(start = 16.dp, end = 16.dp, top = 16.dp, bottom = 8.dp)
+                .height(44.dp)
+                .fillMaxWidth(),
+            elevation = 2.dp,
+            shape = ChatTheme.shapes.inputField
+        ) {
+            SearchInput(
+                modifier = Modifier.fillMaxSize(),
+                query = searchQuery,
+                onValueChange = {
+                    searchQuery = it
+                    listViewModel.setSearchQuery(it)
+                },
+                leadingIcon = { Spacer(Modifier.width(16.dp)) },
+                label = { SearchInputHint() }
+            )
+        }
+
+        ChannelList(
+            modifier = Modifier.fillMaxSize(),
+            viewModel = listViewModel,
+            onChannelClick = onItemClick,
+            emptyContent = { EmptyContent() },
+            itemContent = {
+                Box(
+                    modifier = Modifier
+                        .padding(horizontal = 8.dp)
+                        .fillMaxWidth()
+                        .height(45.dp),
+                ) {
+                    when {
+                        it.isDirectOneToOneChat() -> DirectOneToOneChatItem(
+                            channel = it,
+                            onChannelClick = onItemClick
+                        )
+                        it.isDirectGroupChat() -> DirectGroupChatItem(
+                            channel = it,
+                            onChannelClick = onItemClick
+                        )
+                        else -> ChannelItem(
+                            channel = it,
+                            onChannelClick = onItemClick
+                        )
+                    }
+                }
+            }
+        )
+    }
+}
+
+/**
+ * Component that represents the workspace switch shown in the toolbar.
+ */
+@Composable
+private fun WorkspaceLogo(@DrawableRes logo: Int) {
+    Row {
```

Comment:
I think we can replace this with just an Image that has some padding at the start